tests/test_metrics_parser.py

```diff
@@ -21,3 +23,21 @@ def test_metrics_parser():
 
     # Check that ping names are normalized
     assert "session-end" in parsed_metrics["example.os"]["send_in_pings"]
+
+
+def test_source_url():
+    parser = GleanMetricsParser()
+    parsed_metrics, errs = parser.parse(
+        ["tests/resources/metrics.yaml"], {}, "test.com/foo", "tests"
+    )
+
+    assert (
+        parsed_metrics["example.duration"]["source_url"]
+        == "test.com/foo/blob/tests/resources/metrics.yaml#L4"
```

Comment:
I assume in production, the `repo_url` part includes a `https://` or similar? If it doesn't, maybe we should add it in `get_source_url`? If it does, maybe we should change our example `repo_url` to be more realistic.

Reply:
Yes the `repo_url` from `repositories.yaml` include `https://`. I've updated the test URL to include `https://`